test/common/testpulltask.py

```diff
@@ -120,12 +120,24 @@ def rebase(self, offline=False):
                 master = subprocess.check_output([ "git", "rev-parse", "origin/" + self.base ]).strip()
                 self.sink.status["master"] = master
             subprocess.check_call([ "git", "rebase", "origin/" + self.base ])
-            return None
         except:
             subprocess.call([ "git", "rebase", "--abort" ])
             traceback.print_exc()
             return "Rebase failed"
 
+        # If the bots directory doesn't exist in this branch, check it out from master
+        ret = subprocess.call([ "git", "ls-tree" "-d", "HEAD:bots/"], stdout=devnull, stderr=devnull)
+        if ret == 0:
+            return None
```

Comment:
Nevermind, PEBCAK -- master does have a `bots/` directory with a single README.md. This will still skip checking out master's  bots/ for branches which are in that state, but I suppose there won't/shouldn't be any.